modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,144 @@
+====
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: share-rhel-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a role to grant permission to access the `SharedSecret` object and a `RoleBinding` object that grants permission to access the `SharedSecret` object by running the following command:
++
+[source,terminal]
+----
+$ oc new-project buildah-entitlement
+$ oc apply -f -<<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: use-shared-secret
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - share-entitlement-secret
+    verbs:
+      - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-shared-secret
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: use-shared-secret
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+    namespace: buildah-entitlement
+EOF
+----
+
+. Create a build with relevant mount details using `buildah` buildStrategy:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-rhel-entitlement-build
+spec:
+  source:
+    type: Git
+    git:
+      url: <location of operator's organization>
+      revision: quick-tests
+    contextDir: rh-builds
+  strategy:
+    name: buildah-csi
+    kind: ClusterBuildStrategy
+  paramValues:
+  - name: dockerfile
+    value: Dockerfile
+  volumes:
+  - csi:
+      driver: csi.sharedresource.openshift.io
+      readOnly: true
+      volumeAttributes:
+        sharedSecret: share-entitlement-secret
+    name: rhel-entitlement
+  - name: buildah-auth
+    secret:
+      secretName: buildah-auth
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/buildah-entitlement/sample-go-app
+EOF
+----
+
+. Create the buildRun referencing the rhel-entitlement-build:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: BuildRun
+metadata:
+  name: buildah-rhel-entitlement-buildrun
+spec:
+  build:
+    name: buildah-rhel-entitlement-build
+EOF
+----
+
+[NOTE]
+====
+* You must include the command to remove the `/etc/rhsm-host` directory and all its contents in the Dockerfile before executing any `yum` or `dnf` commands.
+* Use the link:https://access.redhat.com/downloads/content/package-browser[Red Hat Package Browser] to find the correct repositories for your installed packages.
+* You must restore the `/etc/rhsm-host` symbolic link to keep your image compatible with other Red Hat container images.
+* You must set `readOnly` to `true` to mount the shared resource in the build.
+* Reference the name of the `SharedSecret` object to include it in the build. 
```

Comment:
These likewise need to be appropriately footnoted, pointing to an appropriate location in the YAML file. ASCIIDoc supports this.